crates/napi/src/sg_node.rs

```diff
@@ -75,10 +75,15 @@ impl SgNode {
     self.inner.is_named_leaf()
   }
   /// Returns the string name of the node kind
-  #[napi]
+  #[napi(ts_return_type = "T")]
   pub fn kind(&self) -> String {
     self.inner.kind().to_string()
   }
+  /// Check if the node is the same kind as the given `kind` string
+  #[napi]
```

Comment:
I believe napi provides macro to generate type signature so the generating script can be simplified

Reply:
The problem is I couldn't find any macros that allows me to specify type parameters.

I tried to use this feature as much as I could, but in certain cases like the is method, it was not possible.

What I need is `is<K extends T>(kind: K): this is SgNode<K, M>`

It's possible to specify type for the arg `kind: K`, and the return type `SgNode<..>`, but couldn't find a way to do the type parameter `<K extends T>` with napi macros.

Reply:
It's non-blocking and we can figure it out later.

Reply:
napi-rs 3.0 supports function method type overriding but it is not available at the moment. https://github.com/napi-rs/napi-rs/pull/2362